translib/transformer/test/utils_test.go

```diff
@@ -0,0 +1,226 @@
+////////////////////////////////////////////////////////////////////////////////
+//                                                                            //
+//  Copyright 2023 Dell, Inc. 						      //
+//                                                                            //
+//  Licensed under the Apache License, Version 2.0 (the "License");           //
+//  you may not use this file except in compliance with the License.          //
+//  You may obtain a copy of the License at                                   //
+//                                                                            //
+//     http://www.apache.org/licenses/LICENSE-2.0                             //
+//                                                                            //
+//  Unless required by applicable law or agreed to in writing, software       //
+//  distributed under the License is distributed on an "AS IS" BASIS,         //
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  //
+//  See the License for the specific language governing permissions and       //
+//  limitations under the License.                                            //
+//                                                                            //
+////////////////////////////////////////////////////////////////////////////////
+
+package transformer_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"testing"
+
+	. "github.com/Azure/sonic-mgmt-common/translib"
+	db "github.com/Azure/sonic-mgmt-common/translib/db"
+	"github.com/go-redis/redis/v7"
+)
+
+func checkErr(t *testing.T, err error, expErr error) {
+	if err.Error() != expErr.Error() {
+		t.Errorf("Error %v, Expect Err: %v", err, expErr)
+	} else if reflect.TypeOf(err) != reflect.TypeOf(expErr) {
+		t.Errorf("Error type %T, Expect Err Type: %T", err, expErr)
+	}
+}
+
+func processGetRequest(url string, expectedRespJson string, errorCase bool, expErr ...error) func(*testing.T) {
+	return func(t *testing.T) {
+		var expectedMap map[string]interface{}
+		var receivedMap map[string]interface{}
+
+		response, err := Get(GetRequest{Path: url, User: UserRoles{Name: "admin", Roles: []string{"admin"}} })
+		if err != nil {
+			if !errorCase {
+				t.Errorf("Error %v received for Url: %s", err, url)
+			} else if expErr != nil {
+				checkErr(t, err, expErr[0])
+			}
+			return
+		}
+
+		err = json.Unmarshal([]byte(expectedRespJson), &expectedMap)
+		if err != nil {
+			t.Errorf("failed to unmarshal %v err: %v", expectedRespJson, err)
```

Comment:
Use `t.Fatalf()` to stop this test case immediately as there is no point in continuing. `t.Errorf()` marks the test case as failed but does not stop.

Reply:
Addressed as suggested

Reply:
Not addressed

Reply:
Done. Added the corrected file missing in previous commit.